### PR TITLE
Exemplary concrete estimator refactor post interface refactor, of NaiveForecaster

### PR DESCRIPTION
**Pull request by fkiraly**
This is an exemplary refactor of a concrete estimator class, `NaiveForecaster`, to explore how general concrete forecaster refactors would work, along the lines discussed in #912.

This PR changes:

* `_BaseWindowForecaster` still inherits from `BaseForecaster` directly, and already looked extension spec compliant (no overrides, no tags)
* `NaiveForecaster` inherits from `_BaseWindowForecaster`, and has been made extension spec compliant by adding the `requires-fh-in-fit` tag, and moving core logic from `fit` to `_fit`, while avoiding to override `fit`
* it adds one line in `BaseForecaster.fit`, as a general change: the `cutoff` is set to the latest `y` index using `_set_cutoff`. This is done pre-`_fit`, which could override the set cut-off.

### CODE REVIEW

**Comment by fkiraly** [comment]
@mloning, @aiwalter, would one of you mind having at least a superficial look at this? @thayeylolu is following the same pattern in #955, so it would be good to get aligned on "is this good" better earlier than later. Sorry for being pushy...

File: sktime/forecasting/base/_base.py
Comment (mloning) [inline sktime/forecasting/base/_base.py:108]: This is something we forgot in #912, right?

**Review by mloning — APPROVED** [review]
Thanks @fkiraly - this looks good and very easy now given your refactoring in #912.

File: sktime/forecasting/base/_base.py
Comment (mloning) [reply sktime/forecasting/base/_base.py:108]: If so, let's get this merged soon. I has been open for six days already, so I'll merge it later today or tomorrow unless there are any other comments.

Comment (fkiraly) [reply sktime/forecasting/base/_base.py:108]: right, but nothing is failing since the individual forecasters *also* do this currently

Comment (fkiraly) [reply sktime/forecasting/base/_base.py:108]: I also interpret your comment that you'll decide to resolve this convo and/or merge, not me (as the PRer usually would)

Comment (mloning) [reply sktime/forecasting/base/_base.py:108]: Isn't this something that happens in `_set_fh` too? I'll check this later.

Comment (mloning) [reply sktime/forecasting/base/_base.py:108]: Happy for you to merge, but let me check something first. Usually the reviewer merges no?

Comment (mloning) [reply sktime/forecasting/base/_base.py:108]: Looks all good, doesn't happen in `_set_fh`, got confused.

Comment (fkiraly) [reply sktime/forecasting/base/_base.py:108]: ? I thought it´s usually the PRer. See e.g., here:
https://docs.github.com/en/get-started/quickstart/github-flow
"Once your pull request is approved, merge your pull request."

The reason, I guess, is that´s it´s a classical handshake protocol - both PRer and reviewer signal the go-ahead, the reviewer by approving, the PRer by merging. If the reviewer merges, the final handshake is missing a hand.

Comment (fkiraly) [reply sktime/forecasting/base/_base.py:108]: anyway, merged

**Pull request merged**
